winlocalprocessspawner/winlocalprocessspawner.py

```diff
@@ -72,6 +78,14 @@ def get_env(self):
             else:
                 # If the user profile is loaded, adjust APPDATA so the jupyter runtime files are stored
                 # in a per-user location.
+                
+                #If the token is None, the USERPROFILE points at the default directory which is not writable.
+                #this changes the path over to public documents, so at least its a writable location.
+                if 'USERPROFILE' in user_env:

```

Comment:
I believe this can be simplified:

                if 'APPDATA' in user_env:
                    env['APPDATA'] = user_env['APPDATA']
                    env['USERPROFILE'] = user_env['USERPROFILE']
                else:
                    user_env['USERPROFILE'] = user_env['PUBLIC']

if APPDATA is not in user_env, then the system profile was loaded, and we can correctly set the USERPROFILE.

Reply:
Ah, yeah I had noticed the missing APPDATA but I wasn't sure under what other conditions that might be so. I'll give it a try and update the pull.